prov/sm2/src/sm2.h

```diff
@@ -105,7 +105,7 @@ enum {
  * 	sender_gid - id of msg sender
  * 	user_data - the message
  */
-struct sm2_xfer_entry {
+struct sm2_xfer_header {
```

Comment:
I would prefer hdr to match shm/the field name and for shorter names in general but I don't care that match so if that's my only comment, it's ok
also the commit message is a bit off - it says the new struct is called "sm2_inject_header"